File: solutions/java/bob/1/src/main/java/Bob.java
import java.util.regex.Pattern;

public class Bob {

    Bob () {}

    String hey(String phrase) {
        String response = "";

        boolean lc = Pattern.matches("^(.|\n)*[a-z]+(.|\n)*$",phrase);
        boolean e = Pattern.matches(".+! *$",phrase);
        boolean q = Pattern.matches(".+\\? *$",phrase);
        boolean caps = Pattern.matches(".*[A-Z]+.*",phrase);
        boolean och = Pattern.matches(".*[0-9,:{}]+.*",phrase);
        System.out.println(Pattern.matches("^(.|\n)*[a-z]+(.|\n)*$","\nDoes this cryogenic chamber make me look fat?\nno"));
        System.out.println("lc="+lc+",e="+e+",q="+q+",caps="+caps+",och="+och);

        if ((lc || och) && q)response="Sure.";
        else if (!lc && e)response="Whoa, chill out!";
        else if (!lc && caps & !q)response="Whoa, chill out!";
        else if (!lc && caps & q)response = "Calm down, I know what I'm doing!";
        else if (!lc && !e && !q && !caps && !och)response = "Fine. Be that way!";
        else response = "Whatever.";
        return(response);
    }

}

File: solutions/python/bob/1/bob.py
import re
def hey(phrase):
	
	if re.search('^[\t\r\n ]*$',phrase):
		return 'Fine. Be that way!'

	if phrase.upper() == phrase and re.search('^.*[a-zA-Z]+.*\\? *$', phrase):
		return "Calm down, I know what I'm doing!"
		
	if phrase.upper() == phrase and re.search('^.*[a-zA-Z]+.*[^\\?] *$', phrase):
		return "Whoa, chill out!"
			
	if re.search('^.*\\? *$', phrase):
		return 'Sure.' 

	return "Whatever."
